src/controller/user-controller.js

```diff
@@ -0,0 +1,51 @@
+const client = require('../db');
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
+const jwtTokens = require('../utils/jwt-helpers');
+
+
+
+
+const getAllUserHandler = async (req, res) => {
+    try {
+        client.query(`Select * from users`, (err, result)=>{
+            if(!err){
+                res.send(result.rows);   
+            }
+        });
+        client.end;
+    } catch(error) {
+        res.status(500).json({error: error.message})
+    }
+}
+
+const registerHandler =  async (req, res) => {
+    try {
+        const hashedPassword = await bcrypt.hash(req.body.password, 10);
+        const newUser = await client.query(           
+            `INSERT INTO users (user_firstname,user_lastname,user_email,user_password) VALUES ($1,$2,$3,$4) RETURNING *`,
+            [req.body.firstname,req.body.lastname,req.body.email,hashedPassword]);       
+        res.json({user: newUser.rows[0]})
+        client.end;
+    } catch(error){
+        res.status(500).json({error:error.message})
+    }
+}
+
+const loginHandler = async (req, res) => {
+    try {
+        const {email, password} = req.body;
+        const user = await client.query('SELECT * FROM users WHERE user_email = $1',[email])
```

Comment:
the email should be converted to lower case since you are going to be saving in lower case